guides/common/modules/proc_upgrading-the-external-database.adoc

```diff
@@ -1,25 +1,18 @@
-[id="Upgrading_the_External_Database_{context}"]
+[id="Upgrading_the_External_Database{context}"]
 = Upgrading the external database
 
-You can upgrade an external database from {EL} 8 to {EL} 9 while upgrading {Project} from {ProjectVersionPrevious} to {ProjectVersion}.
+The external database will be upgraded when running the {foreman-installer} on the {Project} server.
 
 .Prerequisites
-* Create a new {EL} 9 based host for PostgreSQL server that follows the external database on {EL} 9 documentation.
-ifdef::katello,orcharhino,satellite[]
-For more information, see {InstallingServerDocURL}using-external-databases_{project-context}[Using External Databases with {Project}].
-endif::[]
-* Install PostgreSQL version 13 on the new {EL} host.
+* PostgreSQL version 13 is installed on the {EL} host.
+* The {InstallingServerDocURL}using-external-databases_{project-context}[Using External Databases with {Project}] guide has been followed.
 
 .Procedure
-. Create a backup.
-. Restore the backup on the new server.
-. If {Project} reaches the new database server via the old name, no further changes are required.
-Otherwise reconfigure {Project} to use the new name:
+. Update permissions to prepare the postgresql-evr extension for removal:
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-# {foreman-installer} \
---foreman-db-host _newpostgres.example.com_ \
---katello-candlepin-db-host _newpostgres.example.com_ \
---foreman-proxy-content-pulpcore-postgresql-host _newpostgres.example.com_
+# runuser -l postgres -c \
+"psql -d foreman -c \"UPDATE pg_extension SET extowner = (SELECT oid FROM pg_authid WHERE rolname='foreman') WHERE extname='evr';\""
 ----
+. Follow the {UpgradingDocURL}upgrading_a_connected_{project-context}_server_{context}[Upgrading {Project}] guide.
```

Comment:
In the connected build, this unnecessarily adds an external link instead of an xref. In the disconnected build, this unexpectedly directs users to the connected build.

```suggestion
. Follow xref:upgrading_a_connected_{project-context}_server_{context}[].
```

This _should_ work for the connected build. You might need a different one for disconnected.

Reply:
However, this step also makes me wonder about the sequence of the sections within this assembly/chapter. Looking at the [katello preview](https://theforeman-foreman-documentation-preview-pr-3167.surge.sh/nightly/Upgrading_Project/index-katello.html#Upgrading_the_External_Databaseupgrading-connected): Users start with _2. Upgrading Foreman_, then go through sections _2.1. Foreman server upgrade considerations_, _2.2. Upgrading a connected Foreman server_, _2.3. Performing post-upgrade tasks_, _2.4. Upgrading Smart Proxy servers_, _2.5. Upgrading the external database_... But then 2.5 asks them to follow 2.4 in its last step?

Does this mean that _2.4. Upgrading Smart Proxy servers_ should in fact be the last procedure in _2. Upgrading Foreman_?

Reply:
Looking at the layout of the docs, the upgrade guides here for Foreman/Smart Proxy/External DB don't seem to be in a particular order. Rather, they seem to be their own independent upgrade guides. 

That being said, I think the disconnected DB upgrade part should technically be somewhere in the middle of the Foreman upgrade guide.

Or, it could be viewed as an alternative upgrade guide. If you're using a normal database, follow the regular upgrade guide. If you have an external DB, follow the external DB upgrade guide and then continue to the normal Foreman upgrade guide.